lib/src/queue_list.dart

```diff
@@ -150,6 +150,15 @@ class QueueList<E> extends Object with ListMixin<E> implements Queue<E> {
   @override
   set length(int value) {
     if (value < 0) throw RangeError('Length $value may not be negative.');
+    if (value > length) {
+      try {
+        null as E;
+      } on CastError {
```

Comment:
We might need to double check behavior on dart2js with `-o3`

cc @sigmundch

Reply:
/cc @rakudrama 

I believe in this scenario you are safe, even with -O3, but I worry in general when I see a `catch` for CastError. BTW, I believe in weak-mode this is trivially true (we always allow null, even on non-nullable types)

An alternative w/o try-catch could be to do:
```dart
void Function(T) _foo<T>() => (T a) {};
bool _isNullable<T>() => _foo<T>() is void Function(Null); // True if Null <: T
...
   if (value > length) {
      if (_isNullable<E>()) throw UnsupportedError(...);
```

dart2js may optimize away the cast as trivially true in weak mode, but not in strong mode. The function subtype check is unlikely to be optimized away, but because it doesn't involve try-catch blocks it may end up smaller in strong-mode.

If detecting nullability of a type-parameter is a common pattern, maybe we could expose a common API for it? Or specialize it differently in the compiler?

nit: if you keep the current approach, you should catch `TypeError`, since `CastError` is now deprecated (currently TypeError implements CastError). I'd also recommend creating a helper method in case you end up needing it in a few places, and use `@pragma('dart2js:tryInline')`

Reply:
Agree. Always use `TypeError`, never use `CastError` ever again. There is no `CastError` (eventually).

Reply:
ack